infrastructure/zk/src/format_sql.ts

```diff
@@ -0,0 +1,161 @@
+import * as fs from 'fs';
+import * as utils from './utils';
+import { format } from 'sql-formatter';
+
+function formatQuery(query: string) {
+    let formattedQuery = query;
+    try {
+        formattedQuery = format(query, {
+            language: 'postgresql',
+            tabWidth: 4,
+            keywordCase: 'upper',
+            expressionWidth: 80,
+            indentStyle: 'tabularRight'
+        });
+    } catch {
+        console.error(`Unable to format:\n${query}\n`);
+    }
+
+    // sql-formatter is not smart enough to identify whether something is used as a keyword or a column name
+    const keywordToLower = ['STORAGE', 'TIMESTAMP', 'INPUT', 'DATA', 'ZONE', 'VALUE', 'DEPTH', 'KEY'];
+    for (const keyword of keywordToLower) {
+        // we replace keyword but only if it's not part of a longer word
+        formattedQuery = formattedQuery.replace(RegExp(`\\b${keyword}\\b`, 'g'), keyword.toLowerCase());
+    }
+
+    const formattedLines = formattedQuery.split('\n');
+
+    const minIndent = Math.min(...formattedLines.map((line) => line.search(/\S/)));
+    formattedQuery = formattedQuery
+        .split('\n')
+        .map((line) => line.slice(minIndent))
+        .join('\n');
+
+    return formattedQuery;
+}
+
+function extractQueryFromRustString(query: string): string {
+    query = query.trim();
+    if (query.endsWith(',')) {
+        query = query.slice(0, query.length - 1);
+    }
+    //removing quotes
+    if (!query.startsWith('r#')) {
+        query = query.slice(1, query.length - 1);
+    } else {
+        query = query.slice(3, query.length - 2);
+    }
+
+    //getting rid of all "/" characters, both from escapes and line breaks
+    query = query.replace(/\\/g, '');
+
+    return query;
+}
+
+function embedTextInsideRustString(query: string) {
+    query = query.replace(/"/g, '\\"');
+    const formattedLines = query.split('\n');
+    for (let i = 0; i < formattedLines.length; i++) {
+        let currentLine = '';
+        currentLine += i == 0 ? '"' : ' ';
+        currentLine += formattedLines[i];
+        currentLine += i != formattedLines.length - 1 ? ' \\' : '';
+        currentLine += i == formattedLines.length - 1 ? '"' : '';
+
+        formattedLines[i] = currentLine;
+    }
+    return formattedLines.join('\n');
+}
+
+function addIndent(query: string, indent: number) {
+    return query
+        .split('\n')
+        .map((line) => ' '.repeat(indent) + line)
+        .join('\n');
+}
+
+function formatRustStringQuery(query: string) {
+    const baseIndent = query.search(/\S/);
+    let endedWithAComma = query.trim().endsWith(',');
```

Comment:
- Nit: it seems `endedWithAComma` can be declared as a `const`.
- Bikeshedding: Subjectively, articles (a / the) are usually skipped in identifiers (i.e., `endedWithComma`).
- IMO, it would make sense to remove the comma here so that `extractQueryFromRustString` doesn't need to handle it as well.

Reply:
corrected